ledger/participant-integration-api/src/test/suite/scala/platform/index/InMemoryStateUpdaterSpec.scala

```diff
@@ -51,15 +49,17 @@ class InMemoryStateUpdaterSpec
   behavior of classOf[InMemoryStateUpdater].getSimpleName
 
   "flow" should "correctly process updates" in new Scope {
-    runFlow(Seq(Vector(update1, metadataChangedUpdate) -> 1L, Vector(update3, update4) -> 3L))
-
-    cacheUpdates should contain theSameElementsInOrderAs Seq(
-      Vector(txLogUpdate1),
-      Vector(txLogUpdate3, txRejected),
+    runFlow(
+      Seq(
+        Vector(update1, metadataChangedUpdate) -> 1L,
+        Vector(update3, update4) -> 3L,
+        Vector(update5) -> 4L,
```

Comment:
It's not visible in the assertion what happens with these vectors or their contents, but only the evt seq id. Can you make sure we don't lose entropy here?

Reply:
yeah I switched this PR to draft before I wanted to make the test green :) Need to find a way here

Reply:
I've made an attempt, please have a look